course/models.py

```diff
@@ -474,6 +474,14 @@ def clean(self):
     def is_valid_language(self, lang):
         return lang == "" or lang in [key for key,name in settings.LANGUAGES]
 
+    @property
+    def default_language(self):
+        language = self.language
+        language_code = language.lstrip('|').split('|', 1)[0]
+        if language_code:
+            return language_code
+        return settings.LANGUAGE_CODE.split('-', 1)[0]
```

Comment:
Actually,

```python3
language = self.language
if language and language[0]:
  return language[1:language.find('|', 1)]
return language
```

Seems to be the fastest :/

```
>>> lgs = ('', 'en', '|en|fi|', '|en|fi|sv|')
>>> timeit("for lg in lgs: l = lg.lstrip('|').split('|', 1)[0]", "from __main__ import lgs")
0.5716783219831996
>>> timeit("for lg in lgs: l = lg[1:lg.find('|', 1)] if lg and lg[0] == '|' else lg", "from __main__ import lgs")
0.47175573103595525
```

Reply:
ugh, no need to change because of a small difference. Let's not spend more time on micro-optimization for performance.